test

```diff
@@ -54,12 +68,7 @@ split=(${TEST// / })
 TEST=${split[@]/#/${REPO_PATH}/}
 
 # determine whether target supports race detection
-if [ -z "$GOARCH" ]; then
-	MACHINE_TYPE=$(uname -m)
-	if [ "$MACHINE_TYPE" == "x86_64" ]; then
-		RACE="--race"
-	fi
-elif [ "$GOARCH" == "amd64" ]; then
+if [ "$GOARCH" == "amd64" ] || [[ -z "$GOARCH" && "$MACHINE_ARCH" == "amd64" ]]; then
```

Comment:
this can just be `if [ "$GOARCH" == "amd64" ]; then` if you use the GOARCH suggestion